cloudbuild_public.yaml

```diff
@@ -0,0 +1,66 @@
+steps:
+- name: 'gcr.io/kaniko-project/executor:v1.5.1'
+  id: runtime-image
+  args:
+    [
+      '--cache=true',
+      '--build-arg',
+      'SOURCE_COMMIT=$COMMIT_SHA',
+      '--destination=us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA',
+      '-f', './Dockerfile',
+      '--target=application'
+    ]
+  timeout: 180s
+# CDC end to end tests
+- name: 'gcr.io/$PROJECT_ID/docker-compose'
+  env:
+    - 'CDC_IMAGE=us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA'
+  args:
+    - '-f'
+    - 'docker-compose.gcb.yaml'
+    - run
+    - '--rm'
+    - cdc-e2e-test
+# Clean up after tests
+- name: 'gcr.io/$PROJECT_ID/docker-compose'
+  env:
+    - 'CDC_IMAGE=us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA'
+  args:
+    - '-f'
+    - 'docker-compose.gcb.yaml'
+    - down
+    - '--rmi'
+    - 'local'
+    - '-v'
+# Only tag "nightly" when on master
+- name: 'gcr.io/cloud-builders/docker'
+  secretEnv: ['DOCKER_PASSWORD']
+  entrypoint: 'bash'
+  args:
+  - '-e'
+  - '-c'
+  - |
+    # Only push to Docker Hub from master
+    [ "$BRANCH_NAME" != "master" ] && exit 0
+    docker pull us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA 
+    echo "$$DOCKER_PASSWORD" | docker login --username=sentrybuilder --password-stdin
+    docker tag us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA $PROJECT_ID/$REPO_NAME:$SHORT_SHA
+    docker push $PROJECT_ID/$REPO_NAME:$SHORT_SHA
+    docker tag us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA $PROJECT_ID/$REPO_NAME:$COMMIT_SHA
+    docker push $PROJECT_ID/$REPO_NAME:$COMMIT_SHA
+    docker tag us.gcr.io/$PROJECT_ID/$REPO_NAME:$COMMIT_SHA $PROJECT_ID/$REPO_NAME:nightly
+    docker push $PROJECT_ID/$REPO_NAME:nightly
+images:
```

Comment:
Let's add a note here indicating this is only needed for Freight to find this image like here: https://github.com/getsentry/snuba/blob/5d5a4eb740495df3d54000055b8231643e0c29e5/cloudbuild.yaml#L96